extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson-common/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/JacksonMessageBodyWriterUtil.java

```diff
@@ -0,0 +1,71 @@
+package io.quarkus.resteasy.reactive.jackson.runtime.serialisers;
+
+import static org.jboss.resteasy.reactive.common.providers.serialisers.JsonMessageBodyWriterUtil.setContentTypeIfNecessary;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+
+import javax.ws.rs.core.MultivaluedMap;
+
+import com.fasterxml.jackson.annotation.JsonView;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+
+public final class JacksonMessageBodyWriterUtil {
+
+    private JacksonMessageBodyWriterUtil() {
+    }
+
+    public static ObjectWriter createDefaultWriter(ObjectMapper mapper) {
+        // we don't want the ObjectWriter to close the stream automatically, as we want to handle closing manually at the proper points
+        JsonFactory jsonFactory = mapper.getFactory();
+        if (JacksonMessageBodyWriterUtil.needsNewFactory(jsonFactory)) {
+            jsonFactory = jsonFactory.copy();
+            JacksonMessageBodyWriterUtil.setNecessaryJsonFactoryConfig(jsonFactory);
+            return mapper.writer().with(jsonFactory);
+        } else {
+            return mapper.writer();
+        }
+    }
+
+    private static boolean needsNewFactory(JsonFactory jsonFactory) {
+        return jsonFactory.isEnabled(JsonGenerator.Feature.AUTO_CLOSE_TARGET)
+                || jsonFactory.isEnabled(JsonGenerator.Feature.FLUSH_PASSED_TO_STREAM);
+    }
+
+    public static void setNecessaryJsonFactoryConfig(JsonFactory jsonFactory) {
+        jsonFactory.configure(JsonGenerator.Feature.AUTO_CLOSE_TARGET, false);
+        jsonFactory.configure(JsonGenerator.Feature.FLUSH_PASSED_TO_STREAM, false);
+    }
+
+    public static void doLegacyWrite(Object o, Annotation[] annotations, MultivaluedMap<String, Object> httpHeaders,
```

Comment:
why is it called `doLegacyWrite`?

Reply:
Because it's the "implementation" of the JAX-RS write method, not the new RESTEasy Reactive write method